### PR TITLE
Refactore code-base

### PR DESCRIPTION
**Pull request by Max-Homishchak**
1 - Use Map instead of List for dependency injection 2 - fix transaction type issue, from WITHDRAWAL to DEPOSIT 3 - method refactoring

### CODE REVIEW

File: src/main/java/com/khomishchak/cryptoportfolio/model/enums/ExchangerCode.java [hunk 1]
```diff
@@ -1,7 +1,9 @@
 package com.khomishchak.cryptoportfolio.model.enums;
 
 public enum ExchangerCode {
+    // To be implemented:
     BINANCE,
```
Comment (DendeberiaOleksandr) [inline src/main/java/com/khomishchak/cryptoportfolio/model/enums/ExchangerCode.java:5]: Will this one be deprecated soon?

File: src/main/java/com/khomishchak/cryptoportfolio/services/exchangers/ExchangerServiceImpl.java [hunk 2]
```diff
@@ -48,15 +51,16 @@ public Balance getMainBalance(long userId, ExchangerCode exchangerCode) {
...
     @Override
     public List<Balance> getAllMainBalances(long userId) {
+        List<Balance> balances = new ArrayList<>();
 
-        List<ExchangerCode> codes = userRepository.getReferenceById(userId).getApiKeysSettings()
-                .stream()
-                .map(ApiKeySetting::getCode).toList();
+        userRepository.getReferenceById(userId).getApiKeysSettings().stream()
+                .map(ApiKeySetting::getCode)
+                .forEach(code -> {
```
Comment (DendeberiaOleksandr) [inline src/main/java/com/khomishchak/cryptoportfolio/services/exchangers/ExchangerServiceImpl.java]: brackets can be removed

File: src/main/java/com/khomishchak/cryptoportfolio/services/GoalsServiceImpl.java [hunk 3]
```diff
@@ -83,7 +86,7 @@ public CryptoGoalsTable updateCryptoGoalsTableRecords(List<CryptoGoalsRecordUpda
            CryptoGoalsTableRecord tableRecord = cryptoGoalsTable.getTableRecords().stream()
                    .filter(r -> r.getName().equals(record.ticker()))
                    .findFirst()
-                   .orElseThrow(() -> new RuntimeException("fsa"));
+                   .orElseThrow(() -> new RuntimeException(String.format("Record with ticker:%s was not found for table with id:%d", record.ticker(), tableId)));
```
Comment (DendeberiaOleksandr) [inline src/main/java/com/khomishchak/cryptoportfolio/services/GoalsServiceImpl.java]: It's better to throw specific error e.g `ResourceNotFoundException.java`

File: src/main/java/com/khomishchak/cryptoportfolio/model/enums/ExchangerCode.java (hunk 1 above)
Comment (Max-Homishchak) [reply src/main/java/com/khomishchak/cryptoportfolio/model/enums/ExchangerCode.java:5]: No, those a the exchangers codes that should be implemented

**Review by DendeberiaOleksandr — APPROVED** [review]

**Pull request merged**